src/MarkdownDeep/LinkDefinition.cs

```diff
@@ -58,9 +58,36 @@ internal void RenderLink(Markdown m, StringBuilder b, string link_text, List<str
 			{
 				HtmlTag tag = new HtmlTag("a");
 
+			    var url = this.Url;
+
+			    if (m.DocNetMode && m.ConvertLocalLinks)
+			    {
+			        // A few requirements before we can convert local links:
+                    //   1. Link contains .md
+                    //   2. Link is relative
+                    //   3. Link is included in the index
+                    var index = url.LastIndexOf(".md", StringComparison.OrdinalIgnoreCase);
+			        if (index >= 0)
+			        {
+			            Uri uri;
+			            if (Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out uri))
```

Comment:
```cs
if(Uri.TryCreate(url, UriKind.Relative, out uri))
```

will return true only if the uri is relative, which is what's required. Then the inner test isn't needed.